model-engine/model_engine_server/domain/use_cases/model_endpoint_use_cases.py

```diff
@@ -362,13 +362,22 @@ async def execute(
         # infra_state to make sure that after the update, all resources are valid and in sync.
         # E.g. If user only want to update gpus and leave gpu_type as None, we use the existing gpu_type
         # from infra_state to avoid passing in None to validate_resource_requests.
+        raw_request = request.dict(exclude_unset=True)
```

Comment:
Good Catch here. I'm guessing the bug here was that pydantic fields with non-falsey defaults were evaluating as being set by the user even if they were not explicitly set in the request?

Reply:
yeah, primarily that a user could not specify `None` for fields for which it's allowed, ie `gpu_type`